lxd/include/namespaces.h

```diff
@@ -0,0 +1,17 @@
+#ifndef __NAMESPACES_H
+
+// Define setns() if missing from the C library 
+#ifndef HAVE_SETNS
```

Comment:
This isn't going to work HAVE_SETNS is only present if you use something like autotools which isn't the case here.
So it would always cause the inline definition of setns, failing on any system that does have it.

Reply:
If in your setup, __NR_setns isn't set, then you're also likely not to have setns defined, so you could move the definition of setns under that same check.